ui/components-react/src/components-react/editors/EditorContainer.tsx

```diff
@@ -91,37 +91,52 @@ export interface TypeEditor {
  * EditorContainer React component used by the Tree and PropertyGrid for cell editing.
  * @public
  */
-export class EditorContainer extends React.PureComponent<EditorContainerProps> {
-  private _editorRef: TypeEditor | undefined;
-  private _propertyEditor: PropertyEditorBase | undefined;
-
-  private createEditor(): React.ReactNode {
-    const editorRef = (ref: TypeEditor | undefined) => (this._editorRef = ref);
-
+export function EditorContainer(props: EditorContainerProps) {
+  const {
+    ignoreEditorBlur,
+    title,
+    className,
+    style,
+    shouldCommitOnChange,
+    propertyRecord,
+    setFocus,
+    onCommit,
+    onCancel,
+  } = props;
+
+  let propertyEditor: PropertyEditorBase;
+
+  const editorRef = React.useRef<TypeEditor | null>(null);
+
+  const setEditorRef = (ref: TypeEditor | null) => (editorRef.current = ref);
```

Comment:
Seems like a good place to use `React.useState`

Reply:
I am actually curious about this myself. Could you try `const [editorRef, setEditorRef] = React.useState(React.useRef<TypeEditor | null>(null));` and see if it works fine?

Reply:
Please don't, what's wrong with the ref?

Reply:
Nothing wrong, just thought that we could combine the two things into one